### DISCUSSION TITLE
posts/2022/09/crossorigin-conundrum/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#24] [Announcements]
# posts/2022/09/crossorigin-conundrum/

After days of getting results that I shouldn’t be getting, I’m sending up an SOS to anyone who can set me straight.

https://www.brycewray.com/posts/2022/09/crossorigin-conundrum/

**Comment by davidsneighbour** [comment]
Could it be that having the link-tag in your header right before the actual CSS results in some "Hey please preload this file, no wait here is the CSS please load this file" and it is actually still preloading while the CSS is loading it (again)? Or is the hash maybe different?

I would put everything regarding to load optimisation into header tags, not the `<head>` tags. Outside of the document. 

I can't see any fonts loading in the console, either on your site or mine... might have something broke-configured here in my Chrome. What do the header details show on each of those font-loads? Are they all the same headers? Where in the waterfall are those two calls, both below the css or one below css and one below the page itself? 

You already have preconnect in your headers, so move your preload tags out there too and see if that changes something?

(headers = http request headers, head tag = everything inside `<head>`, I hope I didn't miss any occurence)

**Reply by brycewray** [reply]
@davidsneighbour I should have added that my site doesn’t **currently** have the web fonts, precisely because I kept running into this double-downloading issue and, thus, wanted to avoid it. Sorry.

**Edit**: I have now added a note to the front of the post to clarify the info above.

Thanks for the rest of the answer! I’ll see how that works.

**Comment by brycewray** [comment]
In the meantime, I’ve also started an “Ask HN” thread, hoping the web-font-hating folks on Hacker News will humor me:

https://news.ycombinator.com/item?id=32886945

**Reply by davidsneighbour** [reply]
You probably should add which servertype/hosting you use. I am pretty sure I did all my webfont-optimisation with my own Netlify plugin for Hugo, which is basically a "I thought about it and won't re-think anymore" thing :) But I could dive into the depths... one thing is to not have it all in the HTML but the Headers of the request.

**Reply by brycewray** [reply]
@davidsneighbour Cloudflare Pages, and tested with/without a Cloudflare Worker in front. Same results with each. I may give it a shot on Vercel as well.

As for the headers part, I can’t say I’ve ever seen that approach recommended. Seems MDN *et al.* all opt for the `head` approach I’ve been using. But I’m still looking.

**Reply by davidsneighbour** [reply]
Think about it: Headers are delivered before the page itself is served. You WANT all pre-loading indicators to be received by the browser as early as possible. If "all these sites you check for tips" advise to do it in the HTML they probably serve more people that are unable to add Header configurations to their site. [MDN about Link Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link) usage - it's identical to the `<link>` tag, but it is delivered within the headers, "earlier". 

Also, if you want to read about webfonts, don't forget the "God of web fonts" [Zach Leatherman](https://www.zachleat.com/web/comprehensive-webfonts/).

**Reply by davidsneighbour** [reply]
Sorry, not sure if that came out as clear as it was in my head, in my opinion: If you have a `<link>` tag in your header, you can move it to your HTTP-headers that are delivered before your page is delivered, ... then do it. The `<link>` tag is inferior in my opinion, as it's basically a piece of information to the browser on what to do for the following page. The same (but more obvious) is with the "meta http-equiv" tag. You won't lose any functionality, but you gain the browser knowing things before the actual page starts being parsed.

**Reply by brycewray** [reply]
@davidsneighbour Further testing — now appears this may be specific to CFP, after all; not seeing it with Vercel. I will update the post accordingly once I have some more test results in hand. Thanks again!

*(Update now added.)*
